// react-web-tabs/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _Tab = require('./Tab');

Object.defineProperty(exports, 'Tab', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Tab).default;
  }
});

var _TabComponent = require('./TabComponent');

Object.defineProperty(exports, 'TabComponent', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_TabComponent).default;
  }
});

var _Tabs = require('./Tabs');

Object.defineProperty(exports, 'Tabs', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Tabs).default;
  }
});

var _TabList = require('./TabList');

Object.defineProperty(exports, 'TabList', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_TabList).default;
  }
});

var _TabListComponent = require('./TabListComponent');

Object.defineProperty(exports, 'TabListComponent', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_TabListComponent).default;
  }
});

var _TabPanel = require('./TabPanel');

Object.defineProperty(exports, 'TabPanel', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_TabPanel).default;
  }
});

var _TabPanelComponent = require('./TabPanelComponent');

Object.defineProperty(exports, 'TabPanelComponent', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_TabPanelComponent).default;
  }
});

var _TabProvider = require('./TabProvider');

Object.defineProperty(exports, 'TabProvider', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_TabProvider).default;
  }
});

var _TabSelection = require('./TabSelection');

Object.defineProperty(exports, 'TabSelection', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_TabSelection).default;
  }
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
